parts/k8s/cloud-init/artifacts/mountetcd.sh

```diff
@@ -2,20 +2,26 @@
 # Mounting is done here instead of etcd because of bug https://bugs.launchpad.net/cloud-init/+bug/1692093
 # Once the bug is fixed, replace the below with the cloud init changes replaced in https://github.com/Azure/aks-engine/pull/661.
 set -x
-DISK=/dev/sdc
-PARTITION=${DISK}1
-MOUNTPOINT=/var/lib/etcddisk
+ETCDDISK=""
+PARTITION=""
+for DISK in $(grep -o -G "sd[a-z]" /proc/partitions | uniq); do
+  if ! grep "$DISK"1 /proc/partitions; then
+    if [[ -n $PARTITION ]]; then
```

Comment:
This check is essentially a panic, as the script assumes only a single unpartitioned disk will be found in /proc/partitions on master vms: we deliver a static disk device config:

1. OS disk
2. temp disk
3. 1 data disk (for etcd)

Reply:
Would you want to log some message just so there is more than just a blank exit?
I know this should not get upset but it is a "emergency out" in case of confusion.

`echo "Found more than 1 unpartitioned disk - confused!"`

Reply:
I know it sounds lame but we are trying to be parsimonious with the data payload we send to the Azure cloud-init interface, as there are hard limits. So we can introspect what the heck went wrong here from /var/log/azure/cluster-provision.log

Reply:
Make sense...